src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/Serializer.kt

```diff
@@ -18,6 +22,26 @@ open class Serializer
     private val toStringSerializer = jsonSerializer<Any> { JsonPrimitive(it.src.toString()) }
     private val enumSerializer = jsonSerializer<Any> { JsonPrimitive(serializeEnum(it.src)) }
 
+    private val reportRowViewModelSerializer = jsonSerializer<ReportRowViewModel> {
+
+        val obj = JsonObject()
+
+        for(prop in ReportRowViewModel::class.memberProperties)
+        {
+            if (prop.name != "customFields")
+            {
+                obj.addProperty(convertFieldName(prop.name), prop.get(it.src), it.context)
+            }
+        }
+
+        for(key in it.src.customFields.keys)
```

Comment:
Rather than doing all this custom serialisation to get the rows in the right format for the treetable, I had imagined something like:

`var reports = reports.map(r => ({...r, ...r.customFields}))`

in the javascript to get the array in the desired format.

I feel like this would be simpler, is there a reason that I'm overlooking for handling this on the backend?

Reply:
I guess not, other than it keeps the object a little less bulky. Probably not worth it!